File: function with array.cpp
#include<iostream>
using namespace std;
void scan_array(int a[],int x)
{
    cout<<"enter the elements"<<endl;
    for(int i=0;i<x;i++)
    {
        cin>>a[i];
    }
}
void disp_array(int a[],int x)
{
    cout<<"printing elements"<<endl;
    for(int i=0;i<x;i++)
    {
        cout<<a[i]<<endl;
    }
}
void sum_array(int a[],int x)
{
    int n=0;
  for(int i=0;i<x;i++)
  {
    n = n + a[i];
  }
  cout<<"Sum of elements = "<<n<<endl;
}
void product_array(int a[],int x)
{
    int n=1;
    for(int i=0;i<x;i++)
    {
        n=n*a[i];
    }
    cout<<"product of elements = "<<n<<endl;

}
int main()
{
    int x;
    cout<<"enter the number of elements that you want to enter"<<endl;
    cin>>x;
    int a[x];
    scan_array(a,x);
    disp_array(a,x);
    sum_array(a,x);
    product_array(a,x);
}

File: function with array.java
import java.util.Scanner;
class funct
{
    Scanner s = new Scanner(System.in);
    void scan(int[] a,int x )
    {
      System.out.println("enter the elements\n");
      for(int i=0;i<x;i++)
      {
        a[i]=s.nextInt();
      }
    }
     static void disp(int[] a,int x )
    {
        for(int i=0;i<x;i++)
        {
            System.out.println(a[i]);
        }
        
    }
     static void add(int[] a,int x )
    {
        int p=0;
        for(int i=0;i<x;i++)
        {
            p=p+a[i];
        }
        System.out.println("Sum = "+p);
    }
     static void product(int[] a,int x )
    {
        int p =1;
        for(int i=0;i<x;i++)
        {
           p=p*a[i];
        }
        System.out.println("product = "+p); 
        
    }
}
public class array
{
    public static void main(String a[])
    {
        int x;
        System.out.println("enter the number of elements the you want to enter\n");
        Scanner s = new Scanner(System.in);
        x=s.nextInt();
        int[] arr = new int[x];
        funct f = new funct();
        f.scan(arr,x);
        f.disp(arr,x);
        f.add(arr,x);
        f.product(arr,x);

    }
}